com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -844,35 +975,39 @@ private void Awake()
             }
         }
 
-
         /// <inheritdoc/>
         public override void OnNetworkSpawn()
         {
-            // must set up m_Transform in OnNetworkSpawn because it's possible an object spawns but is disabled
-            //  and thus awake won't be called.
-            // TODO: investigate further on not sending data for something that is not enabled
-            m_Transform = transform;
-            m_ReplicatedNetworkState.OnValueChanged += OnNetworkStateChanged;
-
-            CanCommitToTransform = IsServer;
             m_CachedIsServer = IsServer;
             m_CachedNetworkManager = NetworkManager;
 
+            Initialize();
+
+            // This assures the initial spawning of the object synchronizes all connected clients
+            // with the current transform values. This should not be placed within Initialize since
```

Comment:
Thanks for this comment, warnings like this against moving code around are really nice.